### PR TITLE
handling spotInstancePrice value

### PR DESCRIPTION
**Pull request by lzecca78**
Reopened this PR for some minor fix to be done @emasacco

### CODE REVIEW

File: automated-tests/integration/gcp-gke/cluster.yml [hunk 1]
```diff
@@ -21,3 +21,9 @@ spec:
     maxSize: 1
     volumeSize: 50
     instanceType: n1-standard-1
+  - name: my-spot-node-pool
+    minSize: 1
+    maxSize: 1
+    volumeSize: 50
+    spotInstance: true
```
Comment (angelbarrera92) [inline automated-tests/integration/gcp-gke/cluster.yml:28]: AFAIK, this attribute is wrong. You named it different on the `gke` installer

File: internal/cluster/configuration/gke.go [hunk 2]
```diff
@@ -29,6 +29,7 @@ type GKENodePool struct {
 	MinSize                 int                 `yaml:"minSize"`
 	MaxSize                 int                 `yaml:"maxSize"`
 	InstanceType            string              `yaml:"instanceType"`
+	SpotInstance            bool                `yaml:"spotInstance"`
```
Comment (angelbarrera92) [inline internal/cluster/configuration/gke.go:32]: Same here

File: internal/cluster/provisioners/gke/provisioner.go [hunk 3]
```diff
@@ -128,6 +129,8 @@ func (e GKE) createVarFile() (err error) {
 			buffer.WriteString(fmt.Sprintf("instance_type = \"%v\"\n", np.InstanceType))
 			buffer.WriteString(fmt.Sprintf("max_pods = %v\n", np.MaxPods))
 			buffer.WriteString(fmt.Sprintf("volume_size = %v\n", np.VolumeSize))
+			buffer.WriteString(fmt.Sprintf("preemptible = %v\n", np.SpotInstance))
```
Comment (angelbarrera92) [inline internal/cluster/provisioners/gke/provisioner.go:132]: Be consistent...

Comment (lzecca78) [reply internal/cluster/provisioners/gke/provisioner.go:132]: @angelbarrera92, you asked for a `soft interface` many times, saying that was the added value that SIGHUP put into. I just tried to comply with your requests keeping the same value in both the cloud providers. In the terraform of GKE, I have to maintain the ubiquitous language that google chooses for the name of that kind of instance.
This is why I opted for keeping in the yaml the same `spotInstance` name for both providers (because is a level of indirection) while in the underneath implementation of the terraform I kept the domain right name for both providers.

Comment (angelbarrera92) [reply internal/cluster/provisioners/gke/provisioner.go:132]: Sure!
What I don't agree with is using different attribute names on the `terraform modules`. It triggers exactly these confusions/doubts...

Comment (lzecca78) [reply internal/cluster/provisioners/gke/provisioner.go:132]: the terraform code is on the domain side of the cloud provider, so for respecting the domain ubiquitous language, I must put the exact name that the cloud provider involves.

Comment (angelbarrera92) [reply internal/cluster/provisioners/gke/provisioner.go:132]: You may be right, but if you notice, there are already attributes with standard (our standard) names in terraform modules... like:

- subnetworks: on AWS are subnet-ids while on GKE is a different concept closer to `availability zones`.
- additional_firewall_rules: On AWS are security groups while on GKE are firewall rules.

Then, when we need some specific attributes on gke (for example) we decided to go with the same attribute name from the upstream module plus the `gke_` prefix.

Example: https://github.com/sighupio/furyctl/blob/master/data/provisioners/cluster/gke/variables.tf#L76-L80

TL;DR: What I want is consistency, that's all.

File: automated-tests/integration/gcp-gke/cluster.yml (hunk 1 above)
Comment (nandajavarma) [reply automated-tests/integration/gcp-gke/cluster.yml:28]: made the variable name consistently `spot_instance`

File: data/provisioners/bootstrap/gcp/main.tf [hunk 4]
```diff
@@ -17,7 +17,7 @@ terraform {
 
 
 module "vpc-and-vpn" {
-  source = "github.com/sighupio/fury-gke-installer//modules/vpc-and-vpn?ref=v1.7.0"
+  source = "github.com/sighupio/fury-gke-installer//modules/vpc-and-vpn?ref=v1.7.1"
```
[the remark this answers is not in the document]
Comment (nandajavarma) [reply data/provisioners/bootstrap/gcp/main.tf]: I have updated this for now, I will add a small hook to update it automatically

File: internal/cluster/configuration/gke.go (hunk 2 above)
Comment (nandajavarma) [reply internal/cluster/configuration/gke.go:32]: fixed!

**Comment by nutellinoit** [comment]
To be re-tested

File: .drone.yml [hunk 5]
```diff
@@ -89,84 +92,84 @@ steps:
...
-#      - echo " Our vsphere environment is not so stable"
-#      - echo " The destroy phase fails randomly because of disconnection events between vsphere nodes and the API"
-#      - echo " Ignore errors on this step"
-#      - bats -t ./automated-tests/e2e-tests/vsphere/tests-destroy.sh
+
+  - &e2e
+    name: e2e-gcp
+    image: quay.io/sighup/e2e-furyctl:v1.2.1_v0.2.2_v1.20.1_20.04
+    pull: always
+    privileged: true # Required to connect to the VPN
```
Comment (ralgozino) [inline .drone.yml:100]: If the pipeline runs on drone master, we don't need this anymore
```suggestion

```
This was needed when we had drone hosted on the cloud instead of our vSphere env

**Review by ralgozino — APPROVED** [review]
just a minor change to the drone.yaml file. Feel free to merge afterwards

File: .drone.yml (hunk 5 above)
Comment (nutellinoit) [reply .drone.yml:100]: We need the parameter for the VPN connection on EKS and GKE environments
